gluoncv/data/somethingsomethingv2/classification.py

```diff
@@ -0,0 +1,405 @@
+    this dataset and how to prepare it.
+
+    Parameters
+    ----------
+    root : str, default '~/.mxnet/datasets/somethingsomethingv2'
+        Path to the folder stored the dataset.
+    setting : str, required
+        Config file of the prepared dataset.
+    train : bool, default True
+        Whether to load the training or validation set.
+    test_mode : bool, default False
+        Whether to perform evaluation on the test set
+    name_pattern : str, default None
+        The naming pattern of the decoded video frames.
+        For example, 000012.jpg
+    video_ext : str, default 'mp4'
+        If video_loader is set to True, please specify the video format accordinly.
+    is_color : bool, default True
+        Whether the loaded image is color or grayscale
+    modality : str, default 'rgb'
+        Input modalities, we support only rgb video frames for now.
+        Will add support for rgb difference image and optical flow image later.
+    num_segments : int, default 1
+        Number of segments to evenly divide the video into clips.
+        A useful technique to obtain global video-level information.
+        Limin Wang, etal, Temporal Segment Networks: Towards Good Practices for Deep Action Recognition, ECCV 2016
+    new_length : int, default 1
+        The length of input video clip. Default is a single image, but it can be multiple video frames.
+        For example, new_length=16 means we will extract a video clip of consecutive 16 frames.
+    new_step : int, default 1
+        Temporal sampling rate. For example, new_step=1 means we will extract a video clip of consecutive frames.
+        new_step=2 means we will extract a video clip of every other frame.
+    new_width : int, default 340
+        Scale the width of loaded image to 'new_width' for later multiscale cropping and resizing.
+    new_height : int, default 256
+        Scale the height of loaded image to 'new_height' for later multiscale cropping and resizing.
+    target_width : int, default 224
+        Scale the width of transformed image to the same 'target_width' for batch forwarding.
+    target_height : int, default 224
+        Scale the height of transformed image to the same 'target_height' for batch forwarding.
+    temporal_jitter : bool, default False
+        Whether to temporally jitter if new_step > 1.
+    video_loader : bool, default False
+        Whether to use video loader to load data.
+    use_decord : bool, default True
+        Whether to use Decord video loader to load data. Otherwise use mmcv video loader.
+    transform : function, default None
+        A function that takes data and label and transforms them.
+    """
+    def __init__(self,
+                 setting=os.path.expanduser('~/.mxnet/datasets/somethingsomethingv2/train_videofolder.txt'),
+                 root=os.path.expanduser('~/.mxnet/datasets/somethingsomethingv2/20bn-something-something-v2-frames'),
```

Comment:
seems `setting` is in par with `root`, and `root` is different from doc?

Reply:
Thank you for catching it. Already make `root` the same both in doc and code. 

For the difference between `setting` and `root`, 
`root`: is where the data actually stored
`setting`: is a `.txt` file containing the data path, and used by dataloader.